packages/async-flow/src/types.d.ts

```diff
@@ -0,0 +1,252 @@
+import type { Passable } from '@endo/pass-style';
+import type { Vow } from '@agoric/vow';
+import type { LogStore } from './log-store.js';
+import type { Bijection } from './bijection.js';
+
+export type FlowState =
+  | 'Running'
+  | 'Sleeping'
+  | 'Replaying'
+  | 'Failed'
+  | 'Done';
+
+/**
+ * `T` defaults to `any`, not `Passable`, because unwrapped guests include
+ * non-passables, like unwrapped functions and unwrapped state records.
+ * (Unwrapped functions could be made into Remotables,
+ * but since they still could not be made durable, in this context
+ * it'd be pointless.)
+ */
+export type Guest<T extends unknown = any> = T;
+export type Host<T extends Passable = Passable> = T;
+
+/**
+ * A HostVow must be durably storable. It corresponds to an
+ * ephemeral guest promise.
+ */
+export type HostVow<T extends Passable = Passable> = Host<Vow<T>>;
+
+export type GuestAsyncFunc = (
+  ...activationArgs: Guest[]
+) => Guest<Promise<any>>;
+
+export type HostAsyncFuncWrapper = (...activationArgs: Host[]) => HostVow;
+
+/**
+ * The function from the host as it will be available in the guest.
+ *
+ * Specifically, Vow return values are converted to Promises.
+ */
+export type GuestOf<F extends HostAsyncFuncWrapper> = F extends (
+  ...args: infer A
+) => Vow<infer R>
+  ? (...args: A) => Promise<R>
+  : F;
+
+/**
+ * Convert an entire Guest interface into what the host will implement.
+ */
+type HostInterface<T> = {
+  [K in keyof T]: HostOf<T[K]>;
+};
+
```

Comment:
Nice work on these. Do we also need a `GuestInterface`?

Reply:
When we define an interface it should be in the view of the guest.